src/main/java/com/redhat/red/build/finder/DistributionAnalyzer.java

```diff
@@ -96,47 +105,57 @@ public DistributionAnalyzer(List<File> files, BuildConfig config) {
         return MultiMapUtils.unmodifiableMultiValuedMap(map);
     }
 
+    private String checksum(FileObject fo) throws IOException {
+        return Hex.encodeHexString(DigestUtils.digest(DigestUtils.getDigest(config.getChecksumType().getAlgorithm()), fo.getContent().getInputStream()));
+    }
+
+    private boolean isArchive(FileObject fo) {
+        return (!NON_ARCHIVE_SCHEMES.contains(fo.getName().getExtension()) && Stream.of(sfs.getSchemes()).anyMatch(s -> s.equals(fo.getName().getExtension())));
+    }
+
+    private String normalizePath(FileObject fo) {
+        String friendlyURI = fo.getName().getFriendlyURI();
+        String normalizedPath = friendlyURI.substring(friendlyURI.indexOf(rootString) + rootString.length());
+
+        return normalizedPath;
+    }
+
+    private void listArchive(FileObject fo) throws IOException {
+        LOGGER.debug("Creating file system for: {}", normalizePath(fo));
+
+        try (FileObject layered = sfs.createFileSystem(fo.getName().getExtension(), fo)) {
+            listChildren(layered);
+            sfs.closeFileSystem(layered.getFileSystem());
+        } catch (FileSystemException e) {
+            LOGGER.warn("Unable to process archive/compressed file: {}", red(normalizePath(fo)));
+            LOGGER.debug("Caught file system exception", e);
+        }
+    }
+
     private void listChildren(FileObject fo) throws IOException {
-        FileContent fc = fo.getContent();
-        String friendly = fo.getName().getFriendlyURI();
-        String found = friendly.substring(friendly.indexOf(rootString) + rootString.length());
+        if (!includeFile(fo)) {
+            return;
+        }
 
         if (fo.isFile()) {
-            boolean excludeExtension = config.getArchiveExtensions() != null && !config.getArchiveExtensions().isEmpty() && !config.getArchiveExtensions().stream().anyMatch(x -> x.equals(fo.getName().getExtension()));
-            boolean excludeFile = false;
-
-            if (!excludeExtension) {
-                excludeFile = config.getExcludes() != null && !config.getExcludes().isEmpty() && config.getExcludes().stream().anyMatch(friendly::matches);
+            if (isArchive(fo)) {
```

Comment:
I moved the archive logic (`isArchive()`) inside the `isFile()` block. There didn't seem to be a reason to always check for `isFolder()` when this is the case as it was doing in the previous code.